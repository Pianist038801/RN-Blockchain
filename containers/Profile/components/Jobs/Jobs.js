import React from 'react'
import PropTypes from 'prop-types'
import { View, Text } from 'react-native'
import _ from 'underscore'
import Job from '../Job'
import styles from '../../../../style.js'

const Jobs = (props) => {
  const jobData = props.jobs

  if (!jobData) {
    return (
      <Text>Loading...</Text>
    )
  } else if (jobData.length == 0) {
    return (
      <View>
        <Text style={styles.entityEmptyText}>Yet to be catalogued</Text>
      </View>
    )
  } else {
    const jobs = []
    _.each(jobData, (job) => {
      jobs.push(<Job key={job.id} job={job} />)
    })

    return (
      <View>
        {jobs}
      </View>
    )
  }
}

Jobs.propTypes = {
  jobs: PropTypes.oneOfType([
    PropTypes.object,
    PropTypes.array
  ])
}

export default Jobs
