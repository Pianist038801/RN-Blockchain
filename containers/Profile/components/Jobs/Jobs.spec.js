import React from 'react'
import renderer from 'react-test-renderer'
import Jobs from './index'

describe('containers/Profile/components/Jobs', () => {
  describe('when prop.jobs is null', () => {
    it('renders correctly', () => {
      const props = {
        jobs: null
      }

      const tree = renderer.create(
        <Jobs {...props} />
      ).toJSON()

      expect(tree).toMatchSnapshot()
    })
  })

  describe('when prop.jobs is empty', () => {
    it('renders correctly', () => {
      const props = {
        jobs: []
      }

      const tree = renderer.create(
        <Jobs {...props} />
      ).toJSON()

      expect(tree).toMatchSnapshot()
    })
  })

  describe('when prop.jobs is not empty', () => {
    it('renders correctly', () => {
      const props = {
        jobs: [
          {
            id: 11,
            company: 'Awesome company',
            start_date: '2016-11-04',
            end_date: '2017-01-24',
            position: 'developer',
            verified: true
          },
          {
            id: 12,
            company: 'Applied Blockchain',
            start_date: '2017-11-04',
            end_date: '2018-01-24',
            position: 'developer',
            verified: true
          }
        ]
      }

      const tree = renderer.create(
        <Jobs {...props} />
      ).toJSON()

      expect(tree).toMatchSnapshot()
    })
  })
})
