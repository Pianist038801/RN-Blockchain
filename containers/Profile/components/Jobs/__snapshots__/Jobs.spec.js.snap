// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`containers/Profile/components/Jobs when prop.jobs is empty renders correctly 1`] = `
<View>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "color": "#7e7e7e",
        "fontSize": 10,
      }
    }
  >
    Yet to be catalogued
  </Text>
</View>
`;

exports[`containers/Profile/components/Jobs when prop.jobs is not empty renders correctly 1`] = `
<View>
  <View
    style={
      Object {
        "alignItems": "center",
        "borderColor": "#fff",
        "borderWidth": 1,
        "flex": 0.5,
        "flexDirection": "row",
        "marginBottom": 5,
        "padding": 22.5,
        "width": 600,
      }
    }
  >
    <Image
      source={1}
      style={
        Object {
          "height": 75,
          "resizeMode": "contain",
          "width": 75,
        }
      }
    />
    <View
      style={
        Object {
          "marginLeft": 22.5,
          "width": 360,
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "#569cee",
            "fontWeight": "bold",
          }
        }
      >
        developer
      </Text>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
      >
        Awesome company
      </Text>
    </View>
    <View
      style={Object {}}
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "textAlign": "right",
          }
        }
      >
        2016
      </Text>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "textAlign": "right",
          }
        }
      >
        - 
        2017
      </Text>
    </View>
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "borderColor": "#fff",
        "borderWidth": 1,
        "flex": 0.5,
        "flexDirection": "row",
        "marginBottom": 5,
        "padding": 22.5,
        "width": 600,
      }
    }
  >
    <Image
      source={1}
      style={
        Object {
          "height": 75,
          "resizeMode": "contain",
          "width": 75,
        }
      }
    />
    <View
      style={
        Object {
          "marginLeft": 22.5,
          "width": 360,
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "#569cee",
            "fontWeight": "bold",
          }
        }
      >
        developer
      </Text>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
      >
        Applied Blockchain
      </Text>
    </View>
    <View
      style={Object {}}
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "textAlign": "right",
          }
        }
      >
        2017
      </Text>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "textAlign": "right",
          }
        }
      >
        - 
        2018
      </Text>
    </View>
  </View>
</View>
`;

exports[`containers/Profile/components/Jobs when prop.jobs is null renders correctly 1`] = `
<Text
  accessible={true}
  allowFontScaling={true}
  ellipsizeMode="tail"
>
  Loading...
</Text>
`;
